src/components/CalendarDay.jsx

```diff
@@ -55,10 +55,10 @@ class CalendarDay extends React.PureComponent {
     this.setButtonRef = this.setButtonRef.bind(this);
   }
 
-  componentDidUpdate(prevProps) {
+  componentDidUpdate() {
     const { isFocused, tabIndex } = this.props;
     if (tabIndex === 0) {
-      if (isFocused || tabIndex !== prevProps.tabIndex) {
+      if (isFocused) {
```

Comment:
I think the change here is doing more than desired, and it's not super obvious that this behavior is being done specifically for month navigation clicks. Is it possible to hook into the actual click calls in https://github.com/airbnb/react-dates/blob/e8f8c3c86ce0d6885009068e98c51639d99ff323/src/components/DayPicker.jsx#L476-L479 to set some state information or something in order to validate in the previous if case so that it doesn't try to focus in those cases?